### DISCUSSION TITLE
How to write all annotations to the pdf?

### DISCUSSION THREAD

**Discussion by notuntoward** [RyotaUshio/obsidian-pdf-plus#110] [Q&A]
PDF++ is a fantastic plugin but I can't understand how to use what I think is its most valuable feature: writing annotations to pdfs.

As far as I can tell, I should be able to click "floppy disk" icon at the top of the window:
![image](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/3851891/b4732c88-3fc9-44bd-bc74-89017ce6f2da)

And then all annotations would be written to the pdf.  This does seem to have worked a couple of times, but at the moment, no highlights get written.

As an alternative, I see that I can write to the pdf if I select some text on the pdf, right click, and then choose one of the "write highlight" commands below:

![image](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/3851891/6dc18ff8-4ed2-4231-9300-36840e876a7f)

While this does work, is there no way to simply write all highlights?

**Comment by RyotaUshio** [comment]
The floppy disk icon is a **toggle switch**, meaning that while the button is activated, new highlights will be written into the file.
A highlight is not written **at the moment the icon is clicked**, but is written when you click a color palette or run the "copy link to selection or annotation" command (which I assign a hotkey `Cmd`+`Shift`+`C` to in the demo video below) **after you activate that icon**.

https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/5a4abcf0-f37f-4438-a03c-666d60629e3a

One more thing is that you can have the icon **activated by default** by enabling the following option in the PDF++ settings:

<img width="779" alt="Untitled" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/0f0de935-412f-419a-b15e-239bb8ca1e86">

(update: oops this might be not working now, will look into it -> fixed, will work fine from the next version)

**Reply by notuntoward** [reply]
Ah, I was expecting the write to also happen when I had "select to copy" mode turned on, but it doesn't.  Since the effect within obsidian is *similar to what happens when you click the color button, would it make sense to also make it similar pdf-wise?

**Reply by RyotaUshio** [reply]
Wow I was overlooking it, will be fixed in the next version. Thanks!
